wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/ui/authentication/LoginWithEmailScreen.kt

```diff
@@ -77,9 +79,8 @@ fun LoginWithEmailScreen(
 
         is SignInState.Failure -> {
             loading = false
-            val message = (signInState as? SignInState.Failure)
-                ?.message
-                ?: stringResource(LR.string.error_login_failed)
+            val currentState = signInState as? SignInState.Failure
+            val message = (currentState?.message ?: currentState?.errors?.last()?.message?.let { stringResource(id = it) }) ?: stringResource(id = LR.string.error_login_failed)
```

Comment:
Doing all of this on a single line makes it very hard to follow the logic. Can we break this line up a bit?

Also, correct me if I'm wrong, but I dont' think we need the round brackets around `(currentState?.message ?: currentState?.errors?.last()?.message?.let { stringResource(id = it) })`

If I were writing this, I'd probably do it like this, but this certainly not the only way to go (just sharing as an example, I'm not saying it has to be like this):

```kotlin
val message = currentState?.message
    ?: currentState?.errors?.last()?.message?.let { stringResource(id = it) }
    ?: stringResource(id = LR.string.error_login_failed)
```

Reply:
I did feel like that doing this in one line could make it confusing.

Anyways, considering your approach it looks much better than mine. I tried to make it more concise by writing it up like this:

```kotlin
val message = currentState?.message
                ?: stringResource(
                    currentState?.errors?.last()?.message 
                    ?: LR.string.error_login_failed
                )
```
Does this look good?